### PR TITLE
Silence interval

### PR DESCRIPTION
**Pull request by yger**
Addition of a new pre-processor that can take a list of periods (t_start, t_stop) and blank then with 0, or put some noise. This might be required by people doing micro/macro recordings since they have manually labelled artefacts that do not necessarily saturates (so we can not use blank/clip) but would like to be removed

### CODE REVIEW

**Comment by alejoe91** [comment]
@yger the "silence" mode is the same as the `mode="zeros"` for the `remove_artifact` function, no?
I would just add the `noise` mode to that class instead of making a new preprocessor. What do you think?

**Comment by yger** [comment]
No, because some users want to provide a list of periods (and their sizes can differ), and for now, the remove_artefacts somehow assumes the same ms_before and ms_after for every artefacts, am I wrong?

File: spikeinterface/preprocessing/silence_periods.py [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+    periods are zeroed-out (mode = 'zeros'). This is only recommended 
+    for traces that are centered around zero (e.g. through a prior highpass
+    filter); if this is not the case, you can also fill the periods with noise.
+
+    Parameters
+    ----------
+    recording: RecordingExtractor
+        The recording extractor to remove artifacts from
+    list of lists/arrays
+        One list per segment of tuples (frame_start, rame_stop) to silence
```
Comment (alejoe91) [inline spikeinterface/preprocessing/silence_periods.py:21]:
```suggestion
        One list per segment of tuples (start_frame, end_frame) to silence
```

File: spikeinterface/preprocessing/silence_periods.py [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+    Silence user-defined periods from recording extractor traces. By default, 
+    periods are zeroed-out (mode = 'zeros'). This is only recommended 
+    for traces that are centered around zero (e.g. through a prior highpass
+    filter); if this is not the case, you can also fill the periods with noise.
+
+    Parameters
+    ----------
+    recording: RecordingExtractor
+        The recording extractor to remove artifacts from
+    list of lists/arrays
```
Comment (alejoe91) [inline spikeinterface/preprocessing/silence_periods.py:20]:
```suggestion
    list_periods: list of lists/arrays
```

**Comment by alejoe91** [comment]
@khl02007 maybe this approach will also work for you! So you can silence/remove specific frame ranges without relying on the `remove_artifacts` function :)

**Comment by alejoe91** [comment]
fixes #619

**Review by Antho2422 — APPROVED** [review]

**Review by Antho2422 — APPROVED** [review]

**Pull request closed without merging**